v-next/hardhat/src/internal/cli/telemetry/sentry/init.ts

```diff
@@ -1,118 +1,149 @@
+   */
+  transport: Transport;
 }
 
+const GENERIC_SERVER_NAME = "<user-server>";
+
 /**
- * Initialize Sentry for Node, without performance instrumentation.
+ * This function initializes a custom global sentry reporter/client.
+ *
+ * There are two reasons why we customize it, instead of using the default one
+ * provided by @sentry/node:
+ *   - @sentry/node has an astronomical amount of dependencies -- See https://github.com/getsentry/sentry-javascript/discussions/13846
+ *   - We customize the transport to avoid blocking the main Hardhat process
+ *     while reporting errors.
+ *
+ * Once you initialize the custom global sentry reporter, you can use the usual
+ * `captureException` and `captureMessage` functions exposed by @sentry/core.
+ *
+ * The reason that this uses the global instance of sentry (by calling
+ * initAndBind), is that using the client directly doesn't work with the linked
+ * errors integration.
+ *
+ * Calling `init` will also install the global unhandled rejection and uncaught
+ * exception handlers.
  */
-export async function init(options: InitOptions): Promise<void> {
-  const stackParser = stackParserFromStackParserOptions(
-    createStackParser(nodeStackLineParser(createGetModuleFromFilename())),
+export function init(options: GlobalCustomSentryReporterOptions): void {
+  initAndBind<ServerRuntimeClient, ServerRuntimeClientOptions>(
+    ServerRuntimeClient,
+    {
+      dsn: options.dsn,
+      environment: options.environment,
+      serverName: GENERIC_SERVER_NAME,
+      release: options.release,
+      initialScope: {
+        contexts: {
+          os: {
+            name: os.type(),
+            build: os.release(),
+            version: os.version(),
+          },
+          device: {
+            arch: os.arch(),
+          },
+          runtime: {
+            name: path.basename(process.title),
+            version: process.version,
+          },
+        },
+      },
+      transport: () => options.transport,
+      integrations: [
+        functionToStringIntegration(),
+        contextLinesIntegration(),
+        linkedErrorsIntegration(),
+        nodeContextIntegration(),
+      ],
+      platform: process.platform,
+      stackParser: stackParserFromStackParserOptions(
+        createStackParser(nodeStackLineParser(createGetModuleFromFilename())),
+      ),
+    },
   );
 
-  // NOTE: We do not include most of the default integrations @sentry/node does
-  // because in the main hardhat process, we don't use the default integrations
-  // at all, and they're of limited use in the context of a reporter subprocess.
-  const integrationOptions = {
-    defaultIntegrations: [
-      // Inbound filters integration filters out events (errors and transactions) mainly based on init inputs we never use
-      // Import from @sentry/core if needed
-      // inboundFiltersIntegration(),
-
-      functionToStringIntegration(),
-      linkedErrorsIntegration(),
-      requestDataIntegration(),
```

Comment:
This one is intentionally skipped?

Reply:
Yes, it's for services: https://docs.sentry.io/platforms/javascript/guides/connect/configuration/integrations/requestdata/